src/Stratis.Bitcoin/Utilities/FullNodeExtensions.cs

```diff
@@ -68,28 +69,30 @@ public static async Task RunAsync(this IFullNode node)
         /// <param name="shutdownCompleteMessage">Message to display on the console when the shutdown is complete.</param>
         public static async Task RunAsync(this IFullNode node, CancellationToken cancellationToken, string shutdownMessage, string shutdownCompleteMessage)
         {
-            node.Start();
-
-            if (!string.IsNullOrEmpty(shutdownMessage))
-            {
-                Console.WriteLine();
-                Console.WriteLine(shutdownMessage);
-                Console.WriteLine();
-            }
+            var nodeLifetime = node.Services.ServiceProvider.GetRequiredService<INodeLifetime>() as NodeLifetime;
 
             cancellationToken.Register(state =>
             {
                 ((INodeLifetime)state).StopApplication();
             },
-            node.NodeLifetime);
+            nodeLifetime);
 
             var waitForStop = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
-            node.NodeLifetime.ApplicationStopping.Register(obj =>
+            nodeLifetime.ApplicationStopping.Register(obj =>
             {
                 var tcs = (TaskCompletionSource<object>)obj;
                 tcs.TrySetResult(null);
             }, waitForStop);
 
+            if (!string.IsNullOrEmpty(shutdownMessage))
+            {
+                Console.WriteLine();
+                Console.WriteLine(shutdownMessage);
+                Console.WriteLine();
+            }
+
+            node.Start();
```

Comment:
What was the rationale behind moving this here? Surely the node needs to be started before the shutdown message should be displayed?

Reply:
The message must be displayed before the node is initialized as the collateral initialization is the process we need to interrupt.